### ISSUE TITLE
Exception: Invalid output parameter of type io.StringIO(), use six.StringIO() instead

### ISSUE THREAD

**Issue by alexxxgu** [conan-io/conan#2697]
Hi,
I'm trying to build OpenSSL on Ubuntu for Android using Conan 1.2.0 and getting error:
```
OpenSSL/1.1.0g@conan/stable: WARN: Build folder /root/.conan/data/OpenSSL/1.1.0g/conan/stable/build/39aa0c5b054d7d622fcb086e1d36a2b94f40d8bd
ERROR: OpenSSL/1.1.0g@conan/stable: Error in build() method, line 83
        self.unix_build()
while calling 'unix_build', line 218
        self.run_in_src("./Configure %s %s" % (target, self._get_flags()), win_bash=win_bash)
while calling 'run_in_src', line 103
        self.run(command, win_bash=win_bash, output=buf)
        Exception: Invalid output parameter of type io.StringIO(), use six.StringIO() instead
```

Works on Conan old Conan 1.0.4

**Comment by lasote** [comment]
Hi, I'm going to fix the recipe of OpenSSL and probably we will release in the following days a minor version for Conan to fix this breaking behavior. 
Thanks for reporting

**Comment by lasote** [comment]
Try the OpenSSL recipe now and report, please. Don't forget to clear the package from the local cache or use the `--update` flag in the conan install/create.

**Comment by memsharded** [comment]
Thanks for reporting. Yes, this could be a partial regression. Partial,
because using io.StringIO instead of six one is already kind of broken in
some context.

We are having a look, will try to fix OpenSSL recipe, and will consider
releasing a minor. Thanks!

**Comment by lasote** [comment]
@memsharded your emails are quite slow hahahhahah

**Comment by alexxxgu** [comment]
OpenSSL recipe is ok now, thanks

**Comment by lasote** [comment]
We replaced the exception with a warning message in the released 1.2.1. OpenSSL recipe was fixed, but anyway Conan won't crash anymore for that reason.
